### ISSUE TITLE
Increase in slowness between Netmiko 1.1 and Netmiko 1.2.8

### ISSUE THREAD

**Issue by ktbyers** [ktbyers/netmiko#390]
This slowed down more than I expected -- why?

**Comment by maximumG** [comment]
Just out of curiosity what do you mean about slowness ? In which part of the code ?

I decided to run some time statistic profiling between a test script with netmiko v1.1.0 and netmiko v1.2.8.

The script is as follow

```python
from netmiko.ssh_dispatcher import ConnectHandler

test = {
    'device_type': 'huawei',
    'ip':   'mip',
    'username': 'muser',
    'password': 'mpass',
}

device = ConnectHandler(**test)
print(device.send_command("display version"))
```

You can find the profiling diagram attached.

Netmiko v1.2.8
--------------------
![netmiko-v1 2 8](https://cloud.githubusercontent.com/assets/4469833/23745582/ff2a9900-04b8-11e7-9155-c503e0784514.png)

Netmiko v1.1.0
--------------------
![diagram-v1 1 0](https://cloud.githubusercontent.com/assets/4469833/23745583/ff2c263a-04b8-11e7-980a-be94022f7f61.png)

**Comment by ktbyers** [comment]
The main issue I observed is just against my test devices, the time to do certain actions went from about 8 seconds to about 13 seconds. I test against these devices a lot so I have quite a bit of data on how they behave normally.

I was a bit surprised that it increased by as much as it did.

I think it is probably due to some code I added to improve the reliability of the verifying we are connected process (i.e. Netmiko tries to read some data post login), but I was surprised it increased by as much as it did.
